### DISCUSSION TITLE
Adapter Cloudflare and a good local development story?

### DISCUSSION THREAD

**Discussion by taviks** [sveltejs/kit#2934] [Q&A]
Does anyone know if it's possible yet to have a good development story when using [Adapter Cloudflare](https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare)?

I'd _love_ a way to run https://github.com/cloudflare/wrangler2 and `svelte-kit dev` and have everything "just work." And by "everything" here I mean access to Cloudflare's KV and Durable Objects and of course really fast HMR.

Note: I have in mind here Cloudflare Pages (not Workers, directly).

My attempts so far have gotten me to this: 

```sh
 npx wrangler@beta pages dev .svelte-kit/cloudflare/ --port 3131 -- npx svelte-kit dev --port 3131
```

The problem with this is that file changes do not update what's being served, nor do they update in the browser. Sveltekit (via Vite) is serving things from memory (as I understand it), but wrangler (beta) is serving the built site that lives in `.svelte-kit/cloudflare/`.

**Comment by taviks** [comment]
@lukeed Knowing that you're with Cloudflare and also a contributor here, perhaps you have some insight on this?

**Comment by harlantwood** [comment]
For others following this thread, see also: https://github.com/sveltejs/kit/issues/2966

**Comment by sundaycrafts** [comment]
I've also written the simple solution (or workaround) here. I hope it will be helpful to you.

https://github.com/cloudflare/cloudflare-docs/issues/13062#issuecomment-1986928294
